docs/table/access_table.rst

```diff
@@ -128,28 +128,28 @@ You can get summary information about the table as follows::
 
 If called as a function then you can supply an ``option`` that specifies
 the type of information to return. The built-in ``option`` choices are
-``attributes`` (column attributes, which is the default) or ``stats``
+``'attributes'`` (column attributes, which is the default) or ``'stats'``
 (basic column statistics). The ``option`` argument can also be a list
 of available options::
 
-  >>> t.info('stats')  # doctest: +SKIP
+  >>> t.info('stats')  # doctest: +FLOAT_CMP
   <Table length=5>
-  name mean      std      min max
-  ---- ---- ------------- --- ---
-     a  6.0 4.24264068712   0  12
-     b  7.0 4.24264068712   1  13
-     c  8.0 4.24264068712   2  14
+  name mean        std        min max
+  ---- ---- ----------------- --- ---
+     a  6.0 4.242640687119285   0  12
```

Comment:
See https://github.com/astropy/astropy/pull/12011#discussion_r682524353

Reply:
Formatting the output of `Table.info('stat')` would be simpler, but it doesn't seem to be possible to do that right now. I propose we keep the changes I've already made so that this example could be tested, and deal with the formatting in #12022 instead.

Reply:
You can use offset variations on the array `[2, 4, 4, 4, 5, 5, 7, 9] ` (found it on wikipedia). This has a mean of 5.0 and a std of 2.0. Then the info table will always be looking tidy.

Reply:
This would work, but replacing the initial array would mean that many more examples would have to be replaced, not just the ones with `info('stats')`. On the other hand that might also be the case if we choose to address this in #12022 and also choose to set the format of the other columns.

What do others think?

Reply:
I think we're getting ourselves into a bike shed. As it is, this PR is already a great improvement. Though "ugly," the extra decimal points do reflect real life cases that doctest can run. I am okay with deferring pretty output to #12022 if others think that is the proper way to go.

Reply:
My only real worry is if these tests will pass on a weird architecture or a 32-bit machine (remembering that the problem is in the number of dashes, not the float compare). As long as that is true then I don't mind the example looking messy.

Reply:
And I guess a related comment is that the original version of the docs represents how it used to look since they were definitely copy-pasted from running the commands interactively. So the formatting has changed in the meantime.

Reply:
If the floating points change that much in 32-bit, can we just use ellipsis and call it a day? Is that an option?

Reply:
The problem is in the header -- the number of spaces in the column row and the number of dashes will change if the floating point representation changes by a digit. So ellipsis won't help. I'm happy enough to revert back to skipping the test, and I'd prefer that to the possibility that the doctest will fail under some circumstances.

Reply:
When #12022 is merged then it will become possible to fully control the number of digits in the printed output and therefore the number of spaces and dashes in the table header.

Reply:
As that PR is proving more challenging than initially estimated and this PR will probably be merged first, are there alternatives for now?

Reply:
I would propose to revert the changes and go back to skipping these tests. Perfect is the enemy of good. Apart from this one thing I'm very happy with this PR (and also greatly appreciate the patience of @eerovaher to stick with it!).

Reply:
The difficult part in #12022 was figuring out what the output here should be, but after the discussion there it looks like the best way forward is to apply better default formatting instead of applying column formats. Implementing that is quite simple and the only reason I haven't finalized #12022 already is because if it gets merged first then it would create a conflict with this PR. So the output here can be made better-looking and it can be tested too, the only open question still left is the order in which those two updates should be applied. Merging this PR as is and then changing the appearance of the output in #12022 looks like a simpler way to proceed.

Reply:
OK, let's just merge this as-is and go ahead with #12022, where you rebase on main. That will cause this test to immediately start failing and you can fix it then.

@nstarman - are all your concerns resolved, or close enough?

Reply:
@taldcroft, all the tests succeed here, why would there be any failures after merging?

Reply:
@eerovaher's plan is a good one. Since there's a roadmap to reformat the example, I'm all good here!

Re. to failing tests, I think @taldcroft means this docs example would fail if in #12022 you update the table format, but don't change this example. This example can't slip under the radar and not be fixed.